presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTimeWithTimeZone.java

```diff
@@ -83,6 +117,18 @@ public boolean equals(Object obj)
     @Override
     public String toString()
     {
-        return Instant.ofEpochMilli(millisUtc).atZone(ZoneId.of(timeZoneKey.getId())).format(formatter);
+        // FIXME This is hack that we need to use as the timezone interpretation depends on date (not only on time)
+        // Additionally we cannot just grab current TZ offset from timezoneOffsetReferenceTimestampUtc and use it
+        // as our formatting library will fail to display expected TZ symbol.
+        long currentMillisOfDay =
+                LocalTime.from(
+                        Instant.ofEpochMilli(timezoneOffsetReferenceTimestampUtc)
+                               .atZone(ZoneId.of(TimeZoneKey.UTC_KEY.getId())))
+                        .toNanoOfDay() / 1_000_000L; // nanos to millis
```

Comment:
Ah, sure, Joda dep is not cool (especially that I'd like to see Joda go away in some time:), so what about:

```
/*
 * Equivalent to using Joda: UTC_CHRONOLOGY.millisOfDay().get(timezoneOffsetReferenceTimestampUtc)
 */
long millisInADay = 86400 * 1000; // or TimeUnit.DAYS.toMillis(1) if you wish
currentMillisOfDay = timezoneOffsetReferenceTimestampUtc % millisInADay;
```
?

By naming this constant (and adding an explanatory comment), you deprive it of any magic it had.